rules-reviewed/jboss-eap/websphere/websphere-mq.windup.xml

```diff
@@ -0,0 +1,363 @@
+            com.ibm.websphere.sib.api.jms.JmsTopic
+            com.ibm.websphere.sib.api.jms.JmsQueueConnectionFactory
+            com.ibm.websphere.sib.api.jms.JmsQueue
+            com.ibm.websphere.sib.api.jms.JmsMsgProducer
+            com.ibm.websphere.sib.api.jms.JmsMsgConsumer
+            com.ibm.websphere.sib.api.jms.JmsDestination
+            com.ibm.websphere.sib.api.jms.JmsConnectionFactory
+            com.ibm.msg.client.jms.JmsConnection
+            com.ibm.msg.client.jms.JmsConnectionMetaData
+            com.ibm.msg.client.jms.JmsDestination
+            com.ibm.msg.client.jms.JmsMessageConsumer
+            com.ibm.msg.client.jms.JmsMessageProducer
+            com.ibm.msg.client.jms.JmsQueue
+            com.ibm.msg.client.jms.JmsQueueBrowser
+            com.ibm.msg.client.jms.JmsQueueConnection
+            com.ibm.msg.client.jms.JmsQueueConnectionFactory
+            com.ibm.msg.client.jms.JmsQueueReceiver
+            com.ibm.msg.client.jms.JmsQueueSender
+            com.ibm.msg.client.jms.JmsQueueSession
+            com.ibm.msg.client.jms.JmsSession
+            com.ibm.msg.client.jms.JmsTemporaryQueue
+            com.ibm.msg.client.jms.JmsTemporaryTopic
+            com.ibm.msg.client.jms.JmsTopic
+            com.ibm.msg.client.jms.JmsTopicConnection
+            com.ibm.msg.client.jms.JmsTopicConnectionFactory
+            com.ibm.msg.client.jms.JmsTopicPublisher
+            com.ibm.msg.client.jms.JmsTopicSession
+            com.ibm.msg.client.jms.JmsTopicSubscriber
+            com.ibm.msg.client.jms.JmsXAConnection
+            com.ibm.msg.client.jms.JmsXAConnectionFactory
+            com.ibm.msg.client.jms.JmsXAQueueConnection
+            com.ibm.msg.client.jms.JmsXAQueueConnectionFactory
+            com.ibm.msg.client.jms.JmsXAQueueSession
+            com.ibm.msg.client.jms.JmsXASession
+            com.ibm.msg.client.jms.JmsXATopicConnection
+            com.ibm.msg.client.jms.JmsXATopicConnectionFactory
+            com.ibm.msg.client.jms.JmsXATopicSession
+            com.ibm.jms.JMSBytesMessage
+            com.ibm.jms.JMSMapMessage
+            com.ibm.jms.JMSMessage
+            com.ibm.jms.JMSObjectMessage
+            com.ibm.jms.JMSStreamMessage
+            com.ibm.jms.JMSTextMessage
+        -->
+        <rule id="websphere-mq-62000">
+            <when>
+                <javaclass references="{prefix}{type}"/>
+            </when>
+            <perform>
+                <hint severity="Mandatory" effort="1">
+                    <message>
+                        `{prefix}{type}` is a proprietary WebSphere MQ implementation of standard Java EE JMS interface.
+                        To migrate to JBoss EAP 6, use the Java EE 6 JMS class `javax.jms.{type}`.
+                    </message>
+                </hint>
+            </perform>
+            <where param="prefix"><matches pattern="com\.ibm(\.websphere\.sib\.api|\.msg.client)?\.jms\.(Jms|JMS)[^\.]+"/></where>
```

Comment:
Also, I don't believe you need to escape `.` in character classes:

`[^\.]` should be `[^.]`, but I don't think `\` will really interfere here because it's not a valid java name.